unlock-app/src/hooks/useCrossChainRoutes.ts

```diff
@@ -1,12 +1,23 @@
-import { useQuery } from '@tanstack/react-query'
-import { ethers } from 'ethers'
+import { useQueries, useQuery } from '@tanstack/react-query'
+import { ZeroAddress, ethers } from 'ethers'
 import { useWeb3Service } from '~/utils/withWeb3Service'
 import { Lock } from '~/unlockTypes'
 import { useAuth } from '~/contexts/AuthenticationContext'
 import { purchasePriceFor } from './usePricing'
 import { getReferrer } from '~/utils/checkoutLockUtils'
-import { CrossChainRoute, getCrossChainRoutes } from '~/utils/theBox'
-import { getAccountTokenBalance } from './useAccount'
+import { CrossChainRoute, getCrossChainRoute } from '~/utils/theBox'
+import { networks } from '@unlock-protocol/networks'
+import { BoxEvmChains } from '@decent.xyz/box-common'
+import { Token } from '@unlock-protocol/types'
+
+interface CrossChainRouteWithBalance extends CrossChainRoute {
+  resolvedAt: number
+  userTokenBalance: string
+}
+
+// TheBox returns BigInts as strings with a trailing 'n'
```

Comment:
🤦

Reply:
looking into it, yes its a real issue when parsing serialized values...